Build fails on powerpc: `error: implicit declaration of function 'z7_BranchConv_1_Enc'`

```
--->  Building py314-pylzma
Executing:  cd "/opt/local/var/macports/build/py314-pylzma-8a806050/work/pylzma-0.6.0" && /opt/local/Library/Frameworks/Python.framework/Versions/3.14/bin/python3.14 -m build --no-isolation --wheel --outdir /opt/local/var/macports/build/py314-pylzma-8a806050/work 
* Getting build dependencies for wheel...
/opt/local/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/setuptools/_distutils/dist.py:289: UserWarning: Unknown distribution option: 'tests_require'
  warnings.warn(msg)
/opt/local/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/setuptools/_distutils/dist.py:289: UserWarning: Unknown distribution option: 'test_suite'
  warnings.warn(msg)
/opt/local/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/setuptools/dist.py:759: SetuptoolsDeprecationWarning: License classifiers are deprecated.
!!

        ********************************************************************************
        Please consider removing the following classifiers in favor of a SPDX license expression:

        License :: OSI Approved :: GNU Library or Lesser General Public License (LGPL)

        See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
        ********************************************************************************

!!
  self._finalize_license_expression()
running egg_info
writing pylzma.egg-info/PKG-INFO
writing dependency_links to pylzma.egg-info/dependency_links.txt
writing requirements to pylzma.egg-info/requires.txt
writing top-level names to pylzma.egg-info/top_level.txt
reading manifest file 'pylzma.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
no previously-included directories found matching 'src/sdk.orig'
adding license file 'LICENSE'
writing manifest file 'pylzma.egg-info/SOURCES.txt'
* Building wheel...
/opt/local/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/setuptools/_distutils/dist.py:289: UserWarning: Unknown distribution option: 'tests_require'
  warnings.warn(msg)
/opt/local/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/setuptools/_distutils/dist.py:289: UserWarning: Unknown distribution option: 'test_suite'
  warnings.warn(msg)
/opt/local/Library/Frameworks/Python.framework/Versions/3.14/lib/python3.14/site-packages/setuptools/dist.py:759: SetuptoolsDeprecationWarning: License classifiers are deprecated.
!!

        ********************************************************************************
        Please consider removing the following classifiers in favor of a SPDX license expression:

        License :: OSI Approved :: GNU Library or Lesser General Public License (LGPL)

        See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
        ********************************************************************************

!!
  self._finalize_license_expression()
running bdist_wheel
running build
running build_py
creating build/lib.macosx-10.6-ppc-cpython-314
copying py7zlib.py -> build/lib.macosx-10.6-ppc-cpython-314
running build_ext
adding support for multithreaded compression
building 'pylzma' extension
creating build/temp.macosx-10.6-ppc-cpython-314/src/compat
creating build/temp.macosx-10.6-ppc-cpython-314/src/pylzma
creating build/temp.macosx-10.6-ppc-cpython-314/src/sdk/C
/opt/local/bin/gcc-mp-14 -Os -arch ppc -isysroot/ -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.6.0 -DCOMPRESS_MF_MT=1 -Isrc/sdk/C -I/var/include -I/opt/local/Library/Frameworks/Python.framework/Versions/3.14/include/python3.14 -c src/compat/LzmaCompatDecode.c -o build/temp.macosx-10.6-ppc-cpython-314/src/compat/LzmaCompatDecode.o -Werror=sign-compare
/opt/local/bin/gcc-mp-14 -Os -arch ppc -isysroot/ -DPY_SSIZE_T_CLEAN=1 -DWITH_COMPAT=1 -DPYLZMA_VERSION=0.6.0 -DCOMPRESS_MF_MT=1 -Isrc/sdk/C -I/var/include -I/opt/local/Library/Frameworks/Python.framework/Versions/3.14/include/python3.14 -c src/pylzma/pylzma.c -o build/temp.macosx-10.6-ppc-cpython-314/src/pylzma/pylzma.o -Werror=sign-compare
In file included from src/pylzma/pylzma.c:31:
src/pylzma/pylzma.c: In function 'pylzma_bcj_ppc_convert':
src/pylzma/../sdk/C/Bra.h:14:37: error: implicit declaration of function 'z7_BranchConv_1_Enc'; did you mean 'z7_BranchConv_PPC_Enc'? [-Wimplicit-function-declaration]
   14 | #define Z7_BRANCH_CONV_ENC_2(name)  z7_ ## name ## _Enc
      |                                     ^~~
src/pylzma/../sdk/C/Bra.h:16:37: note: in expansion of macro 'Z7_BRANCH_CONV_ENC_2'
   16 | #define Z7_BRANCH_CONV_ENC(name)    Z7_BRANCH_CONV_ENC_2(BranchConv_ ## name)
      |                                     ^~~~~~~~~~~~~~~~~~~~
src/pylzma/pylzma.c:191:13: note: in expansion of macro 'Z7_BRANCH_CONV_ENC'
  191 |             Z7_BRANCH_CONV_ENC(name)((Byte *) PyBytes_AS_STRING(result), length, 0); \
      |             ^~~~~~~~~~~~~~~~~~
src/pylzma/pylzma.c:204:1: note: in expansion of macro 'DEFINE_BCJ_CONVERTER'
  204 | DEFINE_BCJ_CONVERTER(ppc, PPC);
      | ^~~~~~~~~~~~~~~~~~~~
src/pylzma/../sdk/C/Bra.h:13:37: error: implicit declaration of function 'z7_BranchConv_1_Dec'; did you mean 'z7_BranchConv_PPC_Dec'? [-Wimplicit-function-declaration]
   13 | #define Z7_BRANCH_CONV_DEC_2(name)  z7_ ## name ## _Dec
      |                                     ^~~
src/pylzma/../sdk/C/Bra.h:15:37: note: in expansion of macro 'Z7_BRANCH_CONV_DEC_2'
   15 | #define Z7_BRANCH_CONV_DEC(name)    Z7_BRANCH_CONV_DEC_2(BranchConv_ ## name)
      |                                     ^~~~~~~~~~~~~~~~~~~~
src/pylzma/pylzma.c:193:13: note: in expansion of macro 'Z7_BRANCH_CONV_DEC'
  193 |             Z7_BRANCH_CONV_DEC(name)((Byte *) PyBytes_AS_STRING(result), length, 0); \
      |             ^~~~~~~~~~~~~~~~~~
src/pylzma/pylzma.c:204:1: note: in expansion of macro 'DEFINE_BCJ_CONVERTER'
  204 | DEFINE_BCJ_CONVERTER(ppc, PPC);
      | ^~~~~~~~~~~~~~~~~~~~
error: command '/opt/local/bin/gcc-mp-14' failed with exit code 1

ERROR Backend subprocess exited when trying to invoke build_wheel
Command failed:  cd "/opt/local/var/macports/build/py314-pylzma-8a806050/work/pylzma-0.6.0" && /opt/local/Library/Frameworks/Python.framework/Versions/3.14/bin/python3.14 -m build --no-isolation --wheel --outdir /opt/local/var/macports/build/py314-pylzma-8a806050/work 
Exit code: 1
```

--- a/src/pylzma/pylzma.c
+++ b/src/pylzma/pylzma.c
@@ -198,6 +198,10 @@ pylzma_bcj_##id##_convert(PyObject *self, PyObject *args) \
     return result; \
 }
 
+// Some PowerPC compilers have a builtin define "PPC" that generates invalid
+// code from the "DEFINE_BCJ_CONVERTER" macro.
+#undef PPC
+
 DEFINE_BCJ_CONVERTER(arm, ARM);
 DEFINE_BCJ_CONVERTER(armt, ARMT);
 DEFINE_BCJ_CONVERTER(arm64, ARM64);

